addons/delivery/models/sale_order.py

```diff
@@ -146,7 +146,7 @@ def _get_invoice_status(self):
         for order in self:
             if order.invoice_status in ['no', 'invoiced']:
                 continue
-            order_lines = order.order_line.filtered(lambda x: not x.is_delivery and not x.is_downpayment and not x.display_type)
+            order_lines = order.order_line.filtered(lambda x: not x.is_delivery and not x.is_downpayment and not x.display_type and x.product_uom_qty)
```

Comment:
Shouldn't you just exclude invoiced SO line as a SO line with quantity 0 is considered as `Fully Invoiced`?

```suggestion
            order_lines = order.order_line.filtered(lambda x: not x.is_delivery and not x.is_downpayment and not x.display_type and x.invoice_status != 'invoiced')
```

I think you will have the same issue with your fix if you:
- create a SO with the 2 products with both quantities set to 1
- deliver only one product and create a backorder
- invoice the delivered product
- at that point, the status of the SO will be `Nothing to Invoice`
- add a shipping
=> The status will become `To Invoice`

Reply:
@kitan191 
Yes, you are totally right! Thanks for the review! I made the change accordingly.